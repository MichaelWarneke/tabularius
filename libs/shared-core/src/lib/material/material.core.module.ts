import { NgModule } from '@angular/core';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { LayoutModule } from '@angular/cdk/layout';
import { CommonModule } from '@angular/common';

@NgModule({
  declarations: [],
  imports: [CommonModule, BrowserAnimationsModule, LayoutModule]
})
export class MaterialCoreModule {}
