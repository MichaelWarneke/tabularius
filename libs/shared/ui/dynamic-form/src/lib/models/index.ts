import {
  FormControl,
  ValidatorFn,
  AbstractControlOptions,
  AsyncValidatorFn,
  FormArray
} from '@angular/forms';

export class FormControlBase extends FormControl {
  controlType = '';
  key = '';
  label = '';
  maxWidth = 100;
  constructor(
    formState?: any,
    validatorOrOpts?:
      | ValidatorFn
      | ValidatorFn[]
      | AbstractControlOptions
      | null,
    asyncValidator?: AsyncValidatorFn | AsyncValidatorFn[] | null
  ) {
    super(formState, validatorOrOpts, asyncValidator);
  }
}

export class FormControlTextbox extends FormControlBase {
  controlType = 'input';
  type = '';
  constructor(
    formState?: any,
    validatorOrOpts?:
      | ValidatorFn
      | ValidatorFn[]
      | AbstractControlOptions
      | null,
    asyncValidator?: AsyncValidatorFn | AsyncValidatorFn[] | null
  ) {
    super(formState, validatorOrOpts, asyncValidator);
  }
}

export class FormControlDate extends FormControlBase {
  controlType = 'datepicker';
  type = '';
  constructor(
    formState?: any,
    validatorOrOpts?:
      | ValidatorFn
      | ValidatorFn[]
      | AbstractControlOptions
      | null,
    asyncValidator?: AsyncValidatorFn | AsyncValidatorFn[] | null
  ) {
    super(formState, validatorOrOpts, asyncValidator);
  }
}

export class FormControlCheckbox extends FormControlBase {
  controlType = 'checkbox';
  type = '';
  constructor(
    formState?: any,
    validatorOrOpts?:
      | ValidatorFn
      | ValidatorFn[]
      | AbstractControlOptions
      | null,
    asyncValidator?: AsyncValidatorFn | AsyncValidatorFn[] | null
  ) {
    super(formState, validatorOrOpts, asyncValidator);
  }
}

export class FormControlSelect extends FormControlBase {
  controlType = 'select';
  type = '';
  options?: string[];
  constructor(
    formState?: any,
    validatorOrOpts?:
      | ValidatorFn
      | ValidatorFn[]
      | AbstractControlOptions
      | null,
    asyncValidator?: AsyncValidatorFn | AsyncValidatorFn[] | null
  ) {
    super(formState, validatorOrOpts, asyncValidator);
  }
}

export class FormControlRadioButton extends FormControlBase {
  controlType = 'radio-button';
  type = '';
  options?: string[];
  constructor(
    formState?: any,
    validatorOrOpts?:
      | ValidatorFn
      | ValidatorFn[]
      | AbstractControlOptions
      | null,
    asyncValidator?: AsyncValidatorFn | AsyncValidatorFn[] | null
  ) {
    super(formState, validatorOrOpts, asyncValidator);
  }
}
export class FormControlArray extends FormArray {
  controlType = 'array';
  key = '';
  label = '';
  type = '';
  options?: string[];
  constructor(
    formState?: any,
    validatorOrOpts?:
      | ValidatorFn
      | ValidatorFn[]
      | AbstractControlOptions
      | null,
    asyncValidator?: AsyncValidatorFn | AsyncValidatorFn[] | null
  ) {
    super(formState, validatorOrOpts, asyncValidator);
  }
}
export { Validators } from '@angular/forms';
