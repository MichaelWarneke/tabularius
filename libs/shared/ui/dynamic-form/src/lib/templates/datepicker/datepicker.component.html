<mat-form-field class="tabu-datepicker" [formGroup]="group">
  <input
    matInput
    [matDatepicker]="picker"
    placeholder="{{control?.label}}"
    [formControlName]="control?.key"
  />
  <mat-datepicker-toggle matSuffix [for]="picker"></mat-datepicker-toggle>
  <mat-datepicker #picker></mat-datepicker>
</mat-form-field>
