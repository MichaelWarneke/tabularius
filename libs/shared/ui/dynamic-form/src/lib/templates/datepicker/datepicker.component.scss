.tabu-datepicker {
  padding: 10px;
}
