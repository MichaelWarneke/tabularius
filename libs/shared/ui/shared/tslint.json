{
  "extends": "../../../../tslint.json",
  "rules": {
    "directive-selector": [true, "attribute", "tabuSharedUi", "camelCase"],
    "component-selector": [true, "element", "tabu-shared-ui", "kebab-case"]
  }
}
