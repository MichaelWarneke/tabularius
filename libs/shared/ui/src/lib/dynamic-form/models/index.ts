export { QuestionBase } from './question-base';
export { DropdownQuestion } from './question-dropdown';
export { TextboxQuestion } from './question-textbox';
export { DatepickerQuestion } from './question-datepicker';
