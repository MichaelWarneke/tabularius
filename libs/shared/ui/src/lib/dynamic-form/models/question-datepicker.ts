import { QuestionBase } from './question-base';

export class DatepickerQuestion extends QuestionBase<string> {
  controlType = 'datepicker';
  type: Date;

  constructor(options: any = {}) {
    super(options);
    this.type = options['type'] || '';
  }
}
