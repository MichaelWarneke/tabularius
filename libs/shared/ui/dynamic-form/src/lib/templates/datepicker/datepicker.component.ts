import { Component, OnInit } from '@angular/core';
import { FormGroup } from '@angular/forms';
import { FormControlTextbox } from '../../models';
import { BaseComponent } from '../base.component';

@Component({
  selector: 'tabu-dynamic-form-datepicker',
  templateUrl: './datepicker.component.html',
  styleUrls: ['./datepicker.component.scss']
})
export class DatepickerComponent extends BaseComponent {
  control: FormControlTextbox | null = null;
  group: FormGroup | null = null;

  constructor() {
    super();
  }
}
