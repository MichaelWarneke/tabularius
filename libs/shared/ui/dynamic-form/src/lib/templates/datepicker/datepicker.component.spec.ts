import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { DatepickerComponent } from './datepicker.component';
import { FormGroup, FormControl } from '@angular/forms';
import { FormControlDate } from '../../models';
import { SharedModule } from '@tabularius/shared/ui';
import { MatNativeDateModule } from '@angular/material';
import { By } from '@angular/platform-browser';

describe('DatepickerComponent', () => {
  let component: DatepickerComponent;
  let fixture: ComponentFixture<DatepickerComponent>;
  const key = 'testKey';
  const label = 'testLabel';

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [SharedModule, NoopAnimationsModule, MatNativeDateModule],
      declarations: [DatepickerComponent]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(DatepickerComponent);
    component = fixture.componentInstance;
    component.group = new FormGroup({ testKey: new FormControl('') });
    component.control = new FormControlDate();
    component.control.key = key;
    component.control.label = label;

    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should contain the label', () => {
    const attributes = fixture.debugElement.query(By.css('input')).attributes;
    const text = attributes['placeholder'];
    expect(text).toContain(label);
  });

  it('should contain the formControlName in attributes', () => {
    const attributes = fixture.debugElement.query(By.css('input')).attributes;
    const text = attributes['ng-reflect-name'];
    expect(text).toContain(key);
  });

  it('should render', () => {
    expect(fixture).toMatchSnapshot();
  });
});
