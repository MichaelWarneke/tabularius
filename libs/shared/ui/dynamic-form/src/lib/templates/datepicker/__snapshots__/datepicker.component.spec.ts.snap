// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatepickerComponent should render 1`] = `
<tabu-dynamic-form-datepicker
  control={[Function FormControlDate]}
  group={[Function FormGroup]}
>
  <mat-form-field
    class="tabu-datepicker mat-form-field ng-tns-c9-0 mat-primary mat-form-field-type-mat-input mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-has-label mat-form-field-hide-placeholder ng-untouched ng-pristine ng-valid _mat-animation-noopable"
    ng-reflect-form="[object Object]"
  >
    <div
      class="mat-form-field-wrapper"
    >
      <div
        class="mat-form-field-flex"
      >
        
        
        <div
          class="mat-form-field-infix"
        >
          <input
            aria-haspopup="true"
            aria-invalid="false"
            aria-required="false"
            class="mat-input-element mat-form-field-autofill-control cdk-text-field-autofill-monitored ng-untouched ng-pristine ng-valid"
            id="mat-input-3"
            matinput=""
            ng-reflect-mat-datepicker="[object Object]"
            ng-reflect-name="testKey"
            ng-reflect-placeholder="testLabel"
            placeholder="testLabel"
          />
          <mat-datepicker />
          <span
            class="mat-form-field-label-wrapper"
          >
            
            <label
              aria-owns="mat-input-3"
              class="mat-form-field-label ng-tns-c9-0 mat-empty mat-form-field-empty ng-star-inserted"
              for="mat-input-3"
              id="mat-form-field-label-7"
              ng-reflect-disabled="true"
              ng-reflect-ng-switch="false"
            >
              
              
              testLabel
              
              
            </label>
          </span>
        </div>
        
        <div
          class="mat-form-field-suffix ng-tns-c9-0 ng-star-inserted"
        >
          <mat-datepicker-toggle
            class="mat-datepicker-toggle"
            matsuffix=""
            ng-reflect-datepicker="[object Object]"
            tabindex="-1"
          >
            <button
              aria-haspopup="true"
              aria-label="Open calendar"
              class="mat-icon-button _mat-animation-noopable"
              mat-icon-button=""
              ng-reflect-disabled="false"
              tabindex="0"
              type="button"
            >
              <span
                class="mat-button-wrapper"
              >
                
                <svg
                  class="mat-datepicker-toggle-default-icon ng-star-inserted"
                  fill="currentColor"
                  focusable="false"
                  height="24px"
                  viewBox="0 0 24 24"
                  width="24px"
                >
                  <path
                    d="M19 3h-1V1h-2v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 16H5V8h14v11zM7 10h5v5H7z"
                  />
                </svg>
              </span>
              <div
                class="mat-button-ripple mat-ripple mat-button-ripple-round"
                matripple=""
                ng-reflect-centered="true"
                ng-reflect-disabled="false"
                ng-reflect-trigger="[object HTMLButtonElement]"
              />
              <div
                class="mat-button-focus-overlay"
              />
            </button>
          </mat-datepicker-toggle>
        </div>
      </div>
      
      <div
        class="mat-form-field-underline ng-tns-c9-0 ng-star-inserted"
      >
        <span
          class="mat-form-field-ripple"
        />
      </div>
      <div
        class="mat-form-field-subscript-wrapper"
        ng-reflect-ng-switch="hint"
      >
        
        
        <div
          class="mat-form-field-hint-wrapper ng-tns-c9-0 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
          style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
        >
          
          <div
            class="mat-form-field-hint-spacer"
          />
        </div>
      </div>
    </div>
  </mat-form-field>
</tabu-dynamic-form-datepicker>
`;
